### ISSUE TITLE
Specifying both maxWidth and maxHeight does not respect aspect ratio

### ISSUE THREAD

**Issue by kari** [derhuerst/gulp-scale-images#17]
According to [Sharp documentation](http://sharp.pixelplumbing.com/en/stable/api-resize/#resize) the default resizing strategy is "cover" and it will crop if both maxWidth and maxHeight are defined. To respect the aspect ratio, the resize.js should should use the fit parameter 'inside':

```javascript
task.resize(cfg.maxWidth, cfg.maxHeight || null, {
    withoutEnlargement: !cfg.allowEnlargement,  
    fit: 'inside'
})
```

Also, the default naming in compute-file-name.js has double dots in before the extension because [path.extname already includes the dot](https://nodejs.org/api/path.html#path_path_extname_path) and the join adds another.

**Comment by derhuerst** [comment]
Thanks for reporting.

PR(s) welcome! Please add the fixes as separate commits.

I'm not sure if adding `fit: 'inside'` is a breaking change. 🤔

**Comment by derhuerst** [comment]
Fixed in #18 and #19. Published as `gulp-scale-images@1.2.2`.
